pkg/registry/populator.go

```diff
@@ -29,16 +29,18 @@ type DirectoryPopulator struct {
 	imageDirMap     map[image.Reference]string
 	overwriteDirMap map[string]map[image.Reference]string
 	overwrite       bool
+	permissive      bool
 }
 
-func NewDirectoryPopulator(loader Load, graphLoader GraphLoader, querier Query, imageDirMap map[image.Reference]string, overwriteDirMap map[string]map[image.Reference]string, overwrite bool) *DirectoryPopulator {
+func NewDirectoryPopulator(loader Load, graphLoader GraphLoader, querier Query, imageDirMap map[image.Reference]string, overwriteDirMap map[string]map[image.Reference]string, overwrite, permissive bool) *DirectoryPopulator {
```

Comment:
I don't think we need to propagate this down so long as we always aggregate errors and continue instead of exiting immediately.

Reply:
@njhale I'm trying to think why we might prefer to aggregate errors instead of propagating it all the way down. 
The reason I originally decided to take it all the way down is coz

a) If we have introduced a concept of `permissive` for `opm` it seems natural to have the same concept for our `DirectoryPopulator` too. Basically if some other implementation instead of `opm` want to use `DirectoryPopulator` it should be aware/have access to `permissive` too.

b) If you scale this to n bundles where n is large, it seems more efficient to fail fast if `permissive` isn't enabled instead of churning all the way through.

Reply:
It seems like the error-permissive concept would benefit from a non-permissive error type, since there are error instances that shouldn't be permitted in any case.

```go
type ErrNonPermissive struct {
	err error
}

func foo(l string) error {
	if err = nonPermissiveErrorReturner(l); err != nil {
		return ErrNonPermissive{err}
	}
	if err = permissiveErrorReturner(l); err != nil {
		return err
	}
	return nil
}

func bar(list []string, permissive bool) error {
	var errs []error
	for _, l := range list {
		err := foo(l)
		if err != nil {
			if errors.As(err, &ErrNonPermissive{}) || !permissive {
				return err
			}
			errs = append(errs, err)
		}
	}
	return utilerrors.NewAggregate(errs)
}
```

Then you wouldn't have to propagate `permissive` down as far, and you could have any procedure that returns an error "know" about permissiveness without implementing (too much) of the handling logic itself

Reply:
> If you scale this to n bundles where n is large, it seems more efficient to fail fast if permissive isn't enabled instead of churning all the way through.

fair enough, but these function signatures are already unwieldy as they are today. For the most part, I think we could benefit from not copying all this data around by making `populate` a method of `DirectoryPopulator` (unless it's used by more than one type of populator).

I'm also of the opinion that a lot of this code will get tossed soon anyway, so it may not be super important to keep things pretty now, but we should definitely think about it for the next iteration of operator-registry.